phoenix-core/src/it/java/org/apache/phoenix/end2end/PhoenixRowTimestampFunctionIT.java

```diff
@@ -0,0 +1,398 @@
+        String tableName = createTestData(rowTimestamp, NUM_ROWS);
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            String sql = "SELECT PHOENIX_ROW_TIMESTAMP(), KV1 FROM " + tableName +
+                    " WHERE PHOENIX_ROW_TIMESTAMP() > PK2 ";
+            try (Statement stmt = conn.createStatement()) {
+                ResultSet rs = stmt.executeQuery(sql);
+                int actualCount = 0;
+                while(rs.next()) {
+                    assertTrue(rs.getDate(1).after(new Date(rowTimestamp)));
+                    rs.getString(2);
+                    assertFalse(rs.wasNull());
+                    actualCount++;
+                }
+                assertEquals(NUM_ROWS, actualCount);
+                rs.close();
+            }
+        }
+    }
+
+    @Test
+    // case: All rows selected should have the phoenix_row_timestamp() > date column
+    // Since we used a past date for column PK2
+    // Projected columns should return non null and expected values.
+    public void testSimpleSelectColsWithPhoenixRowTimestampPredicate() throws Exception {
+        long rowTimestamp = EnvironmentEdgeManager.currentTimeMillis() - TS_OFFSET;
+        String tableName = createTestData(rowTimestamp, NUM_ROWS);
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            String sql = "SELECT KV1 FROM " + tableName +
+                    " WHERE PHOENIX_ROW_TIMESTAMP() > PK2 ";
+            try (Statement stmt = conn.createStatement()) {
+                ResultSet rs = stmt.executeQuery(sql);
+                int actualCount = 0;
+                while(rs.next()) {
+                    String kv1Value = rs.getString(1);
+                    assertFalse(rs.wasNull());
+                    assertTrue(kv1Value.substring(0, "KV2_".length()).compareToIgnoreCase("KV1_") == 0);
+                    actualCount++;
+                }
+                assertEquals(NUM_ROWS, actualCount);
+                rs.close();
+            }
+        }
+    }
+
+    @Test
+    // case: Aggregate SQLs work with PhoenixRowTimestamp predicate.
+    public void testSelectCountWithPhoenixRowTimestampPredicate() throws Exception {
+        long rowTimestamp = EnvironmentEdgeManager.currentTimeMillis() - TS_OFFSET;
+        String tableName = createTestData(rowTimestamp, NUM_ROWS);
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            String sql = "SELECT COUNT(*) FROM " + tableName +
+                    " WHERE PHOENIX_ROW_TIMESTAMP() > PK2 ";
+            try (Statement stmt = conn.createStatement()) {
+                ResultSet rs = stmt.executeQuery(sql);
+                while(rs.next()) {
+                    int rowCount = rs.getInt(1);
+                    assertFalse(rs.wasNull());
+                    assertTrue(rowCount == NUM_ROWS);
+                }
+                rs.close();
+            }
+        }
+    }
+
```

Comment:
Can we add a test with a `WHERE PHOENIX_ROW_TIMESTAMP() < TO_DATE("date string')` or comparison with `TO_TIMESTAMP('timestamp string')`?